unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/RuntimeComponentFactory.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using DCL.Components;
+using DCL.Controllers;
+using DCL.Helpers;
+using DCL.Models;
+using UnityEngine;
+
+namespace DCL
+{
+    public interface IRuntimeComponentFactory
+    {
+        IComponent CreateComponent(int classId, object model);
+        void Initialize();
+    }
+
+    public class RuntimeComponentFactory : IRuntimeComponentFactory
+    {
+        private delegate IComponent ComponentBuilder(int classId, object model);
+
+        private Dictionary<int, ComponentBuilder> builders = new Dictionary<int, ComponentBuilder>();
+
+        public IPoolableComponentFactory poolableComponentFactory { get; private set; }
```

Comment:
Question: Should we add this property in the interface definition? I guess it would ease the testing of the entities that use the `IRuntimeComponentFactory` as dependency.

Reply:
It's tempting to do that!. However, having the `PoolableComponentFactory` inside the general factory its an implementation detail. The runtime factory user wouldn't have to know the details of how the components are constructed, if they are poolable or what.